packages/@aws-cdk/aws-ecs/test/fargate/test.fargate-service.ts

```diff
@@ -1,16 +1,96 @@
-import { expect, haveResource, haveResourceLike } from '@aws-cdk/assert';
+import { beASupersetOfTemplate, expect, haveResource, haveResourceLike } from '@aws-cdk/assert';
 import * as appscaling from '@aws-cdk/aws-applicationautoscaling';
 import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
 import * as ec2 from '@aws-cdk/aws-ec2';
 import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
 import * as cloudmap from '@aws-cdk/aws-servicediscovery';
 import * as cdk from '@aws-cdk/core';
+import * as cxapi from '@aws-cdk/cx-api';
 import { Test } from 'nodeunit';
 import * as ecs from '../../lib';
 import { LaunchType } from '../../lib/base/base-service';
 
 export = {
   "When creating a Fargate Service": {
+    "with service id renamed to resource (with feature flag enabled)"(test: Test) {
+      // GIVEN
+      class NestedService extends cdk.Construct {
+        constructor(scope: cdk.Construct, id: string, props: ecs.FargateServiceProps) {
+          super(scope, id);
+
+          new ecs.FargateService(this, "Resource", props);
+        }
+      }
+
+      const app = new cdk.App({
+        context: { [cxapi.ENABLE_CFN_SERVICE_RESOURCE_RENAME]: 'true' }
+      });
+      const stack = new cdk.Stack(app, 'FeatureStack');
+      const vpc = new ec2.Vpc(stack, 'MyVpc', {});
+      const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
+      const taskDefinition = new ecs.FargateTaskDefinition(stack, 'FargateTaskDef');
+
+      taskDefinition.addContainer("web", {
+        image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+      });
+
+      new ecs.FargateService(stack, "Fargate", {
+        cluster,
+        taskDefinition,
+      });
+
+      new NestedService(stack, "Service", {
+        cluster,
+        taskDefinition
+      });
+
+      // THEN
+      expect(stack).to(beASupersetOfTemplate({
+        Resources: {
+          Fargate001516A4: {
+            Type: "AWS::ECS::Service",
+            Properties: {
```

Comment:
Since we only care about the name, is it okay if you leave `Properties` out or empty? Don't want to accidentally assert things that aren't relevant to this particular test.